independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -407,6 +422,14 @@ ValidationContext validate(List<BeanDeploymentValidator> validators,
         return Collections.unmodifiableList(beans);
     }
 
+    public Collection<BeanInfo> getBeansByType(Type type) {
+        var ret = beansByType.get(type);
+        if (ret == null) {
+            return Collections.emptyList();
+        }
+        return Collections.unmodifiableList(ret);
```

Comment:
I wonder if we could make the values in `beansByType` map immutable so that this wrapping is not necessary?